hibernate-core/src/main/java/org/hibernate/resource/transaction/backend/jdbc/internal/JdbcResourceLocalTransactionCoordinatorImpl.java

```diff
@@ -134,6 +134,7 @@ private void afterBeginCallback() {
 		if(this.timeOut > 0) {
 			transactionCoordinatorOwner.setTransactionTimeOut( this.timeOut );
 		}
+		transactionCoordinatorOwner.afterTransactionBegin();
```

Comment:
I added the afterTransactionBegin() call before the observers as in JdbcResourceLocalTransactionCoordinatorImpl#afterCompletionCallback